### DISCUSSION TITLE
How do I just change directions while maintaining the shape of the link?

### DISCUSSION THREAD

**Discussion by james-ep01** [clientIO/joint#2796] [Q&A]
### Introduction

Click the redirect of the link halo to cross the link and click again to return it to its original link shape, and keep clicking to repeat it.
Which option should I apply to the link?

1. First image
![2](https://github.com/user-attachments/assets/1f23e454-f0ba-459b-8b00-f0d976d1471f)

2. cross the link Image
![click](https://github.com/user-attachments/assets/e9c3e79e-982a-4837-a3b0-6be22fd08ecd)

### Steps to reproduce

Below is the link halo code.

const createLinkHalo = function (cellView) {
const cell = cellView.model; 
const options = {
	theme: 'modern',
	cellView: cellView, 
	type: 'toolbar' ,
	clone: (cell, opt) => {
		const clone = cell.clone();
		return clone;
	},
}; 
const halo = new joint.ui.Halo(options);
halo.render();  
};

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
Hi, this is a bug. We should not only swap the `source` and the `target` but also the reverse the order of `vertices`.
Here's a workaround for the time being:
```ts
const h = new Halo({
    cellView: link.findView(paper),
});

h.removeHandle('direction');
h.addHandle({
    name: 'direction',
    position: Halo.HandlePosition.SE,
    events: {
        pointerdown: function (evt, x, y) {
            link.set({
                source: link.target(),
                target: link.source(),
                vertices: link.vertices().reverse()
            });
        }
    },
});

h.render();
```

**Reply by james-ep01** [reply]
I didn't know there was a direction handle. Thank you.

**Reply by kumilingus** [reply]
Sorry, it's currently not in the documentation. Will update the docs soon.
